src/views/List.jsx

```diff
@@ -1,18 +1,41 @@
+import { useState } from 'react';
+
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	const [itemSearch, setItemSearch] = useState('');
+
+	const handleSearchInput = (e) => {
+		setItemSearch(e.target.value.toLowerCase());
+	};
+
+	const handleClear = () => {
+		setItemSearch('');
+	};
+
 	return (
 		<>
-			<p>
-				Hello from the <code>/list</code> page!
-			</p>
+			<form>
+				<label htmlFor="itemSearch">Search your shopping list:</label>
+				<input
+					type="text"
+					id="itemSearch"
+					name="itemSearch"
+					value={itemSearch}
+					onChange={handleSearchInput}
+				/>
+				{itemSearch ? (
+					<button onClick={handleClear} aria-label="clear search field">
+						x
+					</button>
+				) : null}
+			</form>
 			<ul>
-				{data.map((data, i) => (
-					<ListItem 
-					key={i} 
-					name={data.name} 
-					/>
-				))}
+				{data
+					.filter((item) => item.name.toLowerCase().includes(itemSearch))
+					.map((data, i) => (
```

Comment:
Maybe because useMemo is fresh in my brain. haha... Might be a good idea to memorize the filtered data to avoid unnecessary operations on every render?

Reply:
I haven't used it either, but after reading [this](https://www.joshwcomeau.com/react/usememo-and-usecallback/#the-usecallback-hook-5) (Thank you, Tim!). I think this is the place we can use it, but let me know your thoughts. Maybe we can save this for something we can think about for the last couple of weeks.

Reply:
Okie Dokie, I had a look and I was able to implement useMemo!
([this tutorial](https://www.youtube.com/watch?v=aBWwJ4ibpps) was helpful)
Let me know what you think!

Here is the updated search function:
`
const searchedData = useMemo(() => {
		if(!itemSearch) {
			return data;
		} else {
			return data.filter((item) => {
				item.name.toLowerCase().includes(itemSearch);
			})
		}
	}, [data, itemSearch])
`

Here is the updated function to map the list items onto the page:
`
					searchedData.map((data, i) => {
						return <ListItem key={i} name={data.name} />
					})
`

Reply:
I tested this out. I needed to add a return statement inside of the search function, like this:

`const searchedData = useMemo(() => { if(!itemSearch) {
		return data; } else { 
		return data.filter((item) => { 
			return item.name.toLowerCase().includes(itemSearch); 
			}) 
		} 
	}, [data, itemSearch]);`

Once I did that, the search functionality worked normally.